### DISCUSSION TITLE
Does github support creating organizations and accounts using the sdk / api

### DISCUSSION THREAD

**Discussion by Mohammed-Aman-Khan** [community/community#121403] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I am building a SaaS tool for which I am using Supabase auth. We have login with Google, Apple, GitHub and Microsoft logic.

Whenever the user logs in for the first time on our app using any of these ways. Whenever a new user comes to the app, I want my backend apis to create a new organization whenever the user creates a new project on my application I want my backend apis to create a repo under that user's organization. I want to ask the community does GitHub support creating accounts without an email?

Consider the case where a user has logged in with GitHub using the account `mak@example.com`.
I don't want my architecture to use the user's existing GitHub account on the email `mak@example.com`. When the user logs in using the email `mak@example.com` supabase will create a record in it's auth table for the user and I wanna create an anonymous github account for that user using the supabase auth id field (the unique identifier, not the email).
And then I wanna use this new anonymous user's account to create his new org and new repos for his new projects and everything else just like a normal user would do from the GitHub GUI.

I want help on answering these questions:
1. Does GitHub allow creating user's without an email?
2. If yes then how can I exactly do it using the sdk or the api.

Thanks in advance

**Comment by 0xabdulkhaliq** [comment]
Hi there @Mohammed-Aman-Khan 👋🏻, Actually GitHub's API and SDKs do not support creating user accounts or organizations without an email address. 

GitHub requires a valid email address for creating new user accounts and organizations. This is a fundamental requirement for **GitHub's authentication and account management system**.

GitHub does not allow creating user accounts without providing a valid email address. An email address is a mandatory field when creating a new user account on GitHub.

While you cannot create GitHub user accounts or organizations without an email address using the API or SDK, you can create them programmatically by following these steps:

   1. **Creating a new GitHub user account:**
      - Use the GitHub API endpoint `/user/emails` to add a new email address to an existing user account. This requires authentication with a personal access token or an OAuth token.
      - Once you have added a new email address, you can use the `/user` endpoint to create a new user account with the newly added email address.

   2. **Creating a new GitHub organization:**
      - First, you need to have a valid GitHub user account (created as mentioned above or an existing account).
      - Use the `/user/orgs` endpoint to create a new organization. This endpoint requires authentication with a personal access token or an OAuth token associated with the user account.

It's important to note that GitHub's Terms of Service and API usage guidelines should be followed when creating user accounts and organizations programmatically. Additionally, you should consider the privacy implications of creating accounts on behalf of your users without their explicit consent.

An alternative approach could be to **use a third-party authentication provider** like Auth0 or Okta, which allows creating user accounts without relying on GitHub's authentication system. These providers can then be **integrated with GitHub using OAuth or other authentication flows to manage user access and permissions**.
